src/api/app/controllers/webui/project_controller.rb

```diff
@@ -300,6 +302,13 @@ def create
     @project = Project.new(project_params)
     authorize(@project, :create?)
 
+    if Project.deleted?(params[:project][:name])
+      unless params[:restore_option_provided]
```

Comment:
Actually, I thought I had an idea for this but after thinking 10 minutes about it, it seems that it was not well thought enough haha :-D Changed it.